File: 백준/Silver/10828. 스택/스택.py
n = int(input())
stack = []
answer = []
for _ in range(n):
    info = list(input().split())
    if len(info) == 1:
        commend = info[0]
        if commend == 'pop':
            if stack:
                answer.append(stack.pop())
            else:
                answer.append(-1)
        elif commend == 'size':
            answer.append(len(stack))
        elif commend == 'empty':
            if stack:
                answer.append(0)
            else:
                answer.append(1)
        elif commend == 'top':
            if stack:
                answer.append(stack[-1])
            else:
                answer.append(-1)
    else:
        commend, x = info[0], int(info[1])
        stack.append(x)
for x in answer:
    print(x)

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] stack = new int[n];
        int index = 0;
        StringTokenizer st;
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            String commend = st.nextToken();
            int x ;
            if (commend.equals("push")) {
                x = Integer.parseInt(st.nextToken());
                stack[index++] = x;
            } else if (commend.equals("pop")) {
                if (index == 0) System.out.println(-1);
                else {
                    System.out.println(stack[index - 1]);
                    index--;
                }
            } else if (commend.equals("size")) {
                System.out.println(index);
            } else if (commend.equals("empty")) {
                if (index == 0) System.out.println(1);
                else System.out.println(0);
            } else if (commend.equals("top")) {
                if (index == 0) System.out.println(-1);
                else System.out.println(stack[index - 1]);
            }
        }
    }
}
